code/datums/chatmessage.dm

```diff
@@ -125,6 +125,39 @@
 		target.chat_color_darkened = colorize_string(target.name, 0.85, 0.85)
 		target.chat_color_name = target.name
 
+	// SKYRAT ADDITION - Character preference for chatmessages
+	if(ismob(target))
+		var/mob/speaker = target
+		if(speaker.client?.prefs)
+			// Apply the custom color if they want one
+			var/enable_chat_color_player = speaker.client.prefs.read_preference(/datum/preference/toggle/enable_chat_color_player)
```

Comment:
Always put question marks for stuff after client because we can never be sure that the client is still there or if it was destroyed.